package-res/resources/web/pentaho/type/simple.js

```diff
@@ -246,6 +246,27 @@ define([
       },
       //endregion
 
+      toSpec: function(keyArgs) {
+        var spec = {
+          v: this.value,
+          f: this.formatted
+        };
+
+        if(keyArgs){
+          if(!keyArgs.returnFormattedValues && !keyArgs.inlineTypeSpec) {
+            return this.value;
+          }
+          if(keyArgs.inlineTypeSpec) {
+            spec['_'] = this.meta.toSpec();
+          }
+          if(!keyArgs.returnFormattedValues) {
+            delete spec['f'];
```

Comment:
Deleting has a big performance cost and should be avoided when not really needed.